core/server/data/migrations/versions/3.38/07-remove-send-email-when-published-column.js

```diff
@@ -0,0 +1,43 @@
+const logging = require('../../../../../shared/logging');
+const {addColumn, dropColumn} = require('../../../schema/commands');
+
+module.exports = {
+    async up({connection}) {
+        logging.info('Dropping column "send_email_when_published" from table "posts"');
+        await dropColumn('posts', 'send_email_when_published', connection);
+    },
+
+    async down({connection}) {
+        logging.info('Adding column "send_email_when_published" to table "posts"');
+        await addColumn('posts', 'send_email_when_published', connection, {
+            type: 'bool',
+            nullable: true,
+            defaultTo: false
+        });
+
+        const trx = await connection.transaction();
```

Comment:
Maybe this should be a separate migration? :thinking:

Reply:
We've always kept DDL queries separate from DML before. I suppose having it in a separate migration allows for separating db state between migrations but not sure how much that actually buys us 🤔

Reply:
This is only in the `down` command. If we perform a rollback and there's an error when migrating down do we then do a rollforward to get back to the last "good" state?

Reply:
Hmmm, a roll forward would completely drop the column though, which is the last "good" state - it's a weird one!

Reply:
I think I'd lean more towards following existing patterns and keeping the migrations separate until we have a bigger discussion around the implications?

From the `knex-migrator` usage docs:
> Don't mix DDL/DML statements in a migration script. In MySQL DDL statements use implicit commits.

Keeping them separate also reduces cognitive overhead when deciding what to put into a migration 🤔 If there's no good reason any more to keep them separate then we should think about updating the usage docs and making sure everyone knows when it's ok to mix DDL/DML.

Reply:
Cool - sounds good to me :relaxed:

Reply:
I've updated this in the latest here: https://github.com/TryGhost/Ghost/pull/12357/commits/2bb89e9dae655412cdf97e1e81feb5a707443bdd